### ISSUE TITLE
Documentation for running generated archtypes is not correct (for Windows compile)

### ISSUE THREAD

**Issue by ChristianSchenk711** [eclipse-che/che#5043]
I think the documentation in https://eclipse.org/che/docs/assemblies/assembly-lifecycle/index.html is not correct if you are building your archetypes on windows.

It provides this as an example how to start the archetype:

docker run -it --rm --name run-che \
           -v /var/run/docker.sock:/var/run/docker.sock \
           -v /c/tmp:/data \
           -v /c/archetype/assembly/assembly-main/target/eclipse-che-5.4.0/eclipse-che-5.4.0:/assembly \
              eclipse/che:5.4.0 start --skip:nightly"
              
              
But actually this will not work if the archetype was built on windows. It should be 

docker run -it --rm --name run-che \
           -v /var/run/docker.sock:/var/run/docker.sock \
           -v /c/tmp:/data \
           -v /c/archetype/assembly/assembly-main/target:/assembly \
              eclipse/che:5.4.0 start --skip:nightly"

This was my scenario:

Under Win7 I created an archetype like that:

mvn org.apache.maven.plugins:maven-archetype-plugin:2.4:generate -DarchetypeRepository=http://maven.codenvycorp.com/content/groups/public/ -DarchetypeGroupId=org.eclipse.che.archetype -DarchetypeArtifactId=plugin-wizard-archetype      -DarchetypeVersion=5.9.0 -DgroupId=com.test.components  -DartifactId=test -Dversion=6.0.0 -DskipITs -DinteractiveMode=false

and compiled it on the same machine:

cd test
mvn clean install

Now I tried to start the assembly (on a virtualbox in which I mounted /media/ws which contains the archetype folder test):

docker run -it --rm --name bosch_che -v /var/run/docker.sock:/var/run/docker.sock -v /media/ws/test/assembly-che/assembly-main/target/eclipse-che-5.9.0/eclipse-che-5.9.0:/assembly -v /media/ws/chedata:/data eclipse/che-cli:5.9.0 start

I try to create a workspace using the C++ stack. However starting the workspace agent failed with:

[STDERR] /bin/sh: 270: /home/user/che/ws-agent/bin/catalina.sh: Permission denied

I don't get this error if I compile on Ubuntu. 
I don't get the error if I use target instead of target/eclipse-che-5-9-0/eclipse-che-5-9-0 (compiled on windows or ubuntu)

The reason seems to be that the subfolders contain the already expanded version of the assembly, but since this is windows in my case, the scripts are not executable due to wrong attributes.

**Comment by ghost** [comment]
As to docs you may refer to https://github.com/eclipse/che-docs/blob/master/src/main/_docs/assemblies/assemblies-archetype.md

Archetypes have been tested on Win 10 only.

**Comment by ChristianSchenk711** [comment]
First: as far as I understand now, the assembly needs to be bound to a directory that contains the exploded assembly ("bin, conf, lib, .., tomcat"). So the documentation is fine, sorry for that, but then actually I have a real problem, because I still cannot build the archetype on a windows machine.

The page you refer to uses "archetype all" which generates, builds and runs the archetype.
But I would like to split this:
1. use "archetype generate" to get the archetype (which I want to adapt later)
2. use "mvn clean install" on the generated archetype on my windows host
3. run the custom assembly 
( 3. Just like described here: https://eclipse.org/che/docs/assemblies/assembly-lifecycle/index.html under "Run > using the exploded assembly".)

Is this workflow valid? Or is it generally not possible to build the archetypes on Windows machines? Or is it generally a bad idea to build on a Windows machine?

I retested the workflow again (build on a Windows 10 machine this time): same problem:
1./2. seem to work but 3. is were I'm having troubles. I run the archetype with:

docker run -it --rm --name che -v /var/run/docker.sock:/var/run/docker.sock -v /media/ws/test/assembly-che/assembly-main/target/eclipse-che-6.0.0/eclipse-che-6.0.0:/assembly -v /media/ws/chedata:/data eclipse/che-cli:5.9.0 start

and I get:

[STDERR] /bin/sh: 270: /home/user/che/ws-agent/bin/catalina.sh: Permission denied

And as stated above, on windows the file attributes ("x" for .sh) do not end up properly in the ws_agent.tar.gz's.

**Comment by ChristianSchenk711** [comment]
But I also took the assembly-main/target/eclipse-che-6.0.0.tar.gz (built on windows), unzipped it to some location on my Ubuntu and tried to mount it: same problem. (Permission denied).

And I can see the difference in the .tar.gz's created by the windows and linux build:
In the tar.gz created by maven on windows the file attributes of lib > ws-agent.tar > bin > catalina.sh have no execute flag. 
In the tar.gz created by maven on linux the file attributes have the execute flag. 
And that's the root cause. 
So something is going wrong with the build on windows for archetypes IMO.

**Comment by ghost** [comment]
@ChristianSchenk711 on linux you need to take care of file permissions - https://www.eclipse.org/che/docs/assemblies/archetype/index.html#linux

**Comment by ChristianSchenk711** [comment]
I don't think that this is related: adding uid=1000 and gid=999 (this should be the vboxsf group...) ensures that docker can access the folders I shared with virtualbox from windows. But as I said in my previous comment: I even copied the eclipse-che-6.0.0.tar.gz to my linux machine and untared it to a folder on the Ubuntu file system (nothing shared with windows any more).  And still this problem persists. And as I said: the tar.gz files generated by the build on Linux and Windows are different: the file attributes in the ws-agent.tar inside the eclipse-che-6.0.0.tar.gz are different. And this is not a runtime problem but a build-time problem. Actually if I compare the output from the windows and linux build using Beyond Compare, it looks identical (because Beyond Compare doesn't know about the linux attributes in the tar's). But if I go into the files using 7zip, inside the tar I can see the difference:
Build with linux: test\assembly-che\assembly-main\target\eclipse-che-6.0.0.tar.gz\eclipse-che-6.0.0.tar\eclipse-che-6.0.0\tomcat\bin\catalina.sh is -rwxrwxrwx.
Build with windows its: -rw--r--r--

**Comment by ghost** [comment]
@ChristianSchenk711 I do not have any issues when I run and compile on Linux. 

Once again, we have never tested archetypes on Win 7 and any systems running boot2docker. Also, scenarios like "generate on Win, run on Unix" were not prioritized either.

**Comment by ChristianSchenk711** [comment]
As I said in a comment above, I tested the same on Windows 10 as well. Same problem. So not Win7 related. From the documentation it just seems to be that a lot of people are building on Windows 10 hosts and running with boot2docker or on VMs, so I thought that this is a standard scenario.
But if this is not supported, I cannot help it. For production it anyway makes sense to do everything on a Linux machine.
Just in case anybody else stumbles across that:
I worked around that problem by modifying the assembly.xml files of the server and main assemblies to include <fileMode>0777</fileMode> for all dependencySet's and fileSet's. This solved the problem for me.
